src/app/[locale]/incident/summary/components/IncidentSummaryForm.tsx

```diff
@@ -62,23 +70,36 @@ const IncidentSummaryForm = () => {
         incident_date_start: new Date().toISOString(),
         extra_properties: formState.extra_properties,
       })
-      .then((res) => {
-        if (formState.attachments.length > 0) {
-          const signalId = res.signal_id
-          if (signalId) {
-            formState.attachments.forEach((attachment) => {
-              const formData = new FormData()
-              formData.append('signal_id', signalId)
-              formData.append('file', attachment)
-              postAttachments(signalId, formData)
-            })
+
+      if (formState.attachments.length > 0) {
+        const signalId = res.signal_id
+        if (signalId) {
+          try {
+            await Promise.all(
```

Comment:
Ik ben niet 100% zeker ervan; maar ik denk dat deze await Promise.all() met daarin een map en dan in die map weer een await onnodig is en zelfs slomere code oplevert dan wanneer je de await in de map weghaalt.

Daarnaast wordt er niks in de map gereturned dus zou Promise.all() een lege array terug krijgen of een lijst met undefined items, vermoed ik.

```await Promise.all(formState.attachments.map((attachment) => {
  const formData = new FormData();
  formData.append('signal_id', signalId);
  formData.append('file', attachment);
  return postAttachments(signalId, formData);
}));
```

Deze code doet exact hetzelfde en zou ervoor zorgen dat die Promise.all() ervoor zorgt dat elk item wordt geresolved in plaats van dat je dat zelf weer binnen je array map doet.

Voor de zekerheid misschien nog handig als of @bartjkdp of @Robbert hier naar kijkt?

Reply:
postAttachments returned een promise toch? en we moeten wachten tot alle attachments succesvol of niet zijn afgehandeld met doorgaan naar de volgende pagina. Volgens mij is het wel goed zo hoor.

Reply:
```
await Promise.all(
              formState.attachments.map((attachment) => {
                const formData = new FormData()
                formData.append('signal_id', signalId)
                formData.append('file', attachment)
                return postAttachments(signalId, formData)
              })
```